### PR TITLE
Fix image upload for Liberty and newer

### PR DESCRIPTION
**Pull request by dirkmueller**
--copy-from was deprecated in Kilo and got removed in Liberty, so
we need to upload the image now differently

### CODE REVIEW

File: scripts/qa_crowbarsetup.sh [hunk 1]
```diff
@@ -2872,18 +2872,26 @@ function oncontroller_testsetup()
...
+            'openstack image show $imageid | grep active &>/dev/null' \
+            "prepare cnftools image"
         pushd /var/lib/openstack-tempest-test
         echo 1 > /proc/sys/kernel/sysrq
+        [ -x /usr/bin/tempest-cleanup ] && /usr/bin/tempest-cleanup --init-saved-state || :
         ./run_tempest.sh -N $tempestoptions 2>&1 | tee tempest.log
         tempestret=${PIPESTATUS[0]}
+        [ -x /usr/bin/tempest-cleanup ] && /usr/bin/tempest-cleanup --delete-tempest-conf-objects || :
         testr last --subunit | subunit-1to2 > tempest.subunit.log
         /var/lib/openstack-tempest-test/bin/tempest_cleanup.sh || :
```
Comment (vuntz) [inline scripts/qa_crowbarsetup.sh]: Should we remove that call? (and stop generating this script)

**Comment by vuntz** [comment]
@dirkmueller reading the code (so not using liberty), --copy-from seems to be still there when using v1 API, which seems to be the default. It sounds to me like the v2 API is just incomplete from that perspective -- the plan was to use the taskflows for importing images in general, including copying from an URL, but this bit is not fully implemented yet.

It's probably no big deal for mkcloud, but in general, it's quite suboptimal to download and re-upload...

File: scripts/qa_crowbarsetup.sh (hunk 1 above)
Comment (dirkmueller) [inline scripts/qa_crowbarsetup.sh]: @vuntz yes, on todo (somewhere). removing it right now is sort of difficult as tempest-cleanup doesn't exist for cloud4.

**Comment by dirkmueller** [comment]
@vuntz glanceclient and openstack client default to v2. with liberty there is v2.0, v2.1, v2.3 and 1.0. 

yes, we could also upload the image using v1 api, but I guess we probably rather want to remove v1 asap..

**Comment by vuntz** [comment]
Thanks for the answer. I'm probably misreading the code, but I agree it's not really relevant here anyway.

+1

**Comment by aplanas** [comment]
+1: Tested in c16 with Liberty (tempest deployed)

**Pull request merged**